### DISCUSSION TITLE
Encode kong request header with base64

### DISCUSSION THREAD

**Discussion by Vamsi0473** [Kong/kong#9850] [Help] [answered]
Hi Team,
We have requirement to pass request header which contains chinese language and kong was returning 400 bad request and on troubleshooting we came to see when we encode the chinese characters and send to kong it is working so now we want to transform the request header to base64 encode using request transformer advanced plugin
Thanks,
Vamsi

**Comment by StarlightIbuki** [comment, marked as answer]
By "Chinese language" I suppose you mean Chinese characters encoded with UTF-8 or other normal encodings.
According to [rfc7230](https://www.rfc-editor.org/rfc/rfc7230#ref-USASCII:~:text=field%2Dvchar%20%3D%20VCHAR%20/%20obs%2Dtext)(see p83), bytes other than visible ASCII character cannot be used in header values. Obviously encoded Chinese characters include bytes that are not in this range.
Kong is also acting as an HTTP proxy so it's expected to reject such requests.
